wls-ergebnismeldung-service/src/test/java/de/muenchen/oss/wahllokalsystem/ergebnismeldungservice/service/awerte/AWerteServiceTest.java

```diff
@@ -0,0 +1,186 @@
+            Mockito.when(aWerteModelMapper.fromListOfAWerteEntityToListOfAWerteModel(null)).thenReturn(null);
+            Mockito.when(exceptionFactory.createTechnischeWlsException(ExceptionConstants.GETAWERTE_UNSAVEABLE))
+                    .thenReturn(mockedWlsException);
+
+            Assertions.assertThatThrownBy(() -> unitUnderTest.getAWerte(wahlbezirkID)).isSameAs(mockedWlsException);
+        }
+    }
+
+    @Nested
+    class InitialiseAWerte {
+
+        @Test
+        void should_notFailAndSaveInDB_when_clientDataFound() {
+            val wahlbezirkID1 = "wahlbezirkID1";
+            val aWerteModelListFromClient1 = createListOfAWerteModels(wahlbezirkID1);
+            val wahlbezirkID2 = "wahlbezirkID2";
+            val aWerteModelListFromClient2 = createListOfAWerteModels(wahlbezirkID2);
+            val wahlbezirkID3 = "wahlbezirkID3";
+            val aWerteModelListFromClient3 = createListOfAWerteModels(wahlbezirkID3);
+            val wahlbezirkIDs = List.of(wahlbezirkID1, wahlbezirkID2, wahlbezirkID3);
+            Mockito.when(aWerteClient.getAWerte(wahlbezirkID1)).thenReturn(aWerteModelListFromClient1);
+            Mockito.when(aWerteClient.getAWerte(wahlbezirkID2)).thenReturn(aWerteModelListFromClient2);
+            Mockito.when(aWerteClient.getAWerte(wahlbezirkID3)).thenReturn(aWerteModelListFromClient3);
+
+            Assertions.assertThatNoException().isThrownBy(() -> unitUnderTest.initialiseAWerte(wahlbezirkIDs));
+            Mockito.verify(aWerteRepository, times(3)).saveAll(Mockito.any());
+        }
+
+        @Test
+        void should_notFailAndSaveInDB_when_base64ClientDataFound() {
+            val wahlbezirkID1 = Base64.getEncoder().encodeToString("WAHLBEZIRK-1410\",\"wahlterminId".getBytes());
+            val aWerteModelListFromClient1 = createListOfAWerteModels(wahlbezirkID1);
+            val wahlbezirkID2 = Base64.getEncoder().encodeToString("WAHLBEZIRK-1411\",\"wahlterminId".getBytes());
+            val aWerteModelListFromClient2 = createListOfAWerteModels(wahlbezirkID2);
+            val wahlbezirkID3 = Base64.getEncoder().encodeToString("WAHLBEZIRK-1412\",\"wahlterminId".getBytes());
+            val aWerteModelListFromClient3 = createListOfAWerteModels(wahlbezirkID3);
+            val wahlbezirkIDs = List.of(wahlbezirkID1, wahlbezirkID2, wahlbezirkID3);
+            Mockito.when(aWerteClient.getAWerte(wahlbezirkID1)).thenReturn(aWerteModelListFromClient1);
+            Mockito.when(aWerteClient.getAWerte(wahlbezirkID2)).thenReturn(aWerteModelListFromClient2);
+            Mockito.when(aWerteClient.getAWerte(wahlbezirkID3)).thenReturn(aWerteModelListFromClient3);
+
+            Assertions.assertThatNoException().isThrownBy(() -> unitUnderTest.initialiseAWerte(wahlbezirkIDs));
+            Mockito.verify(aWerteRepository, times(3)).saveAll(Mockito.any());
+        }
+
+        @Test
+        void should_notFailAndLogError_when_clientDataFoundAndRepoFails() {
```

Comment:
dürfte verständlicher sein

```suggestion
        void should_notFailButLogError_when_clientDataFoundAndRepoFails() {
```